static/js/import_wizard/utils/detection.ts

```diff
@@ -0,0 +1,216 @@
+        };
+        mValValidated.column = col;
+      }
+    }
+
+    // If key "placeType" is present then the value must be of type DCType.
+    // Note, if the validation succeeds, the value (DCType) is replaced by a map of
+    // columnIdx (number) to the DCType.
+    if (
+      _.includes(Object.keys(mVal), PLACETYPE_API_KEY) &&
+      mVal[PLACETYPE_API_KEY]
+    ) {
+      if (!isDCTypeOrProperty(mVal[PLACETYPE_API_KEY])) {
+        continue;
+      }
+      // If made this far, then mVal must also have had the column field. Extract the columnIdx
+      // and assign it as a key to mVal["placeProperty"].
+      if (_.isEmpty(col)) {
+        continue;
+      }
+      mValValidated.placeType = {
+        [col.columnIdx]: {
+          dcid: mVal[PLACETYPE_API_KEY][DCID_API_KEY],
+          displayName: mVal[PLACETYPE_API_KEY][DISP_NAME_API_KEY],
+        },
+      };
+    }
+
+    // If key "placeProperty" is present then the value must be of type DCProperty.
+    // Note, if the validation succeeds, the value (DCProperty) is replaced by a map of
+    // columnIdx (number) to the DCProperty.
+    if (
+      _.includes(Object.keys(mVal), PLACEPROP_API_KEY) &&
+      mVal[PLACEPROP_API_KEY]
+    ) {
+      if (!isDCTypeOrProperty(mVal[PLACEPROP_API_KEY])) {
+        continue;
+      }
+      // If made this far, then mVal must also have had the column field. Extract the columnIdx
+      // and assign it as a key to mVal["placeProperty"].
+      if (_.isEmpty(col)) {
+        continue;
+      }
+      mValValidated.placeProperty = {
+        [col.columnIdx]: {
+          dcid: mVal[PLACEPROP_API_KEY][DCID_API_KEY],
+          displayName: mVal[PLACEPROP_API_KEY][DISP_NAME_API_KEY],
+        },
+      };
+    }
+
+    // If key "headers" is present, then the value must be an array for Columns.
+    if (_.includes(Object.keys(mVal), HEADER_API_KEY) && mVal[HEADER_API_KEY]) {
+      if (!Array.isArray(mVal[HEADER_API_KEY])) {
+        continue;
+      }
+      const headers: Array<Column> = [];
+      for (const c of mVal[HEADER_API_KEY]) {
+        if (isColumn(c)) {
+          headers.push({
+            id: c[COLUMN_ID_API_KEY],
+            header: c[COLUMN_HEADER_API_KEY],
+            columnIdx: c[COLUMN_IDX_API_KEY],
+          });
+        }
+      }
+      if (headers.length > 0) {
+        mValValidated.headers = headers;
+      }
+    }
+    if (!_.isEmpty(mValValidated)) {
+      mToReturn[key] = mValValidated;
+    }
+  }
+  return mToReturn;
+}
+
+/**
+ * Given the supported DC types and Properties (for Places) response
+ * from the detection server API, parse it in to an array of TypeProperty
+ * objects.
+ *
+ * @param supportedTypesAPIResponse is the Object response from the server API.
+ *
+ * @returns an array of TypeProperty objects. If there are any unexpected
+ * validation errors, that particularTypeProperty is skipped.
+ */
+export function parseSupportedTypePropertiesResponse(
+  supportedTypesAPIResponse
```

Comment:
super nit: when camel casing acronyms, they should be treated as a word, i.e. supportedTypesApiResponse

Reply:
Done.